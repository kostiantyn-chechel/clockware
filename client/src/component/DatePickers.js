import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import { today } from '../helpers/dateTime';

const useStyles = makeStyles((theme) => ({
    container: {
        display: 'flex',
        flexWrap: 'wrap',
    },
    textField: {
        marginLeft: theme.spacing(1),
        marginRight: theme.spacing(1),
        width: 200,
    },
}));

export default function DatePickers(props) {
    const classes = useStyles();

    return (
        <div className={classes.container} >
            <TextField
                onChange={props.onChange}
                id="date"
                label="Дата:"
                type="date"
                defaultValue={today()}
                className={classes.textField}
                InputLabelProps={{
                    shrink: true,
                }}
            />
        </div>
    );
}