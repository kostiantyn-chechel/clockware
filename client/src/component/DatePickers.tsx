import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import { today } from '../helpers/dateTime';

const useStyles = makeStyles((theme) => ({
    container: {
        display: 'flex',
        flexWrap: 'wrap',
    },
    textField: {
        marginLeft: theme.spacing(1),
        marginRight: theme.spacing(1),
        width: 200,
    },
}));

interface IDatePickers {
    onChange(): void;
}

const DatePickers:React.FC<IDatePickers> = (props) => {
    const classes = useStyles();

    return (
        <div className={classes.container} >
            <TextField
                onChange={props.onChange}
                id="date"
                label="Дата:"
                type="date"
                defaultValue={today()}
                className={classes.textField}
                InputLabelProps={{
                    shrink: true,
                }}
            />
        </div>
    );
};

export default DatePickers;