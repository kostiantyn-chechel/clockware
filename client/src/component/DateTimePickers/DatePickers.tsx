import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import { today } from '../../helpers/dateTime';

const useStyles = makeStyles((theme) => ({
    container: {
        display: 'flex',
        flexWrap: 'wrap',
    },
    textField: {
        marginLeft: theme.spacing(1),
        marginRight: theme.spacing(1),
        width: 200,
    },
}));

interface IDatePickers {
    onChange(event: React.ChangeEvent<{ value: string }>): void;
    defaultDate?: string;
}

const DatePickers:React.FC<IDatePickers> = (props) => {
    const classes = useStyles();

    const defDate = () => {
        if (props.defaultDate) return props.defaultDate;
        return today();
    };

    return (
        <div className={classes.container} >
            <TextField
                onChange={props.onChange}
                id="date"
                label="Дата:"
                type="date"
                // defaultValue={today()}
                defaultValue={defDate()}
                className={classes.textField}
                InputLabelProps={{
                    shrink: true,
                }}
            />
        </div>
    );
};

export default DatePickers;