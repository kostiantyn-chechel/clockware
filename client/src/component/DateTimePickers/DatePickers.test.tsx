import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import DatePickers from './DatePickers';

describe('Test <DatePickers>', () => {
    const onChangeMock = jest.fn();
    const defDateMock = jest.fn(() => '2021-01-20');

    const Component = <DatePickers onChange={onChangeMock}/>;
    const ShallowComponent = shallow(Component);

    test('DatePickers component should exists', () => {
        expect(ShallowComponent).toBeDefined();
    });

    // test("ToMatchSnapshot DatePickers with default date", () => {
    //     expect(renderer.create(Component).toJSON()).toMatchSnapshot();
    // });

    test('ToMatchSnapshot DatePickers with date in props', () => {
        const Component = <DatePickers onChange={onChangeMock} defaultDate='2021-01-01'/>;
        expect(renderer.create(Component).toJSON()).toMatchSnapshot();
    });

    test("DatePickers changes value after click", () => {
        const MountComponent = mount(Component);
        const newDate = '2021-02-01';

        expect(onChangeMock).toHaveBeenCalledTimes(0);

        MountComponent.find("input").simulate("change", {
            target: { value: newDate },
        });

        expect(onChangeMock).toHaveBeenCalledTimes(1);

        // expect(onChangeMock).toHaveBeenCalledWith(newDate); //???
    });

});