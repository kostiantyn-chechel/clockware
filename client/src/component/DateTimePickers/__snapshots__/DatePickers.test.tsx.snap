// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <DatePickers> ToMatchSnapshot DatePickers with date in props 1`] = `
<div
  className="makeStyles-container-1"
>
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-textField-2"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink={true}
      htmlFor="date"
      id="date-label"
    >
      Дата:
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        defaultValue="2021-01-01"
        disabled={false}
        id="date"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="date"
      />
    </div>
  </div>
</div>
`;
